### ISSUE TITLE
macos: ERRO[0000] finding config on system: unable to resolve HOME directory: user: unknown userid 501

### ISSUE THREAD

**Issue by ssbarnea** [containers/podman#11126]
I found a bug on macos which should not happen. Try to unset `HOME` and run `podman info`, you will see something like
```
ERRO[0000] finding config on system: unable to resolve HOME directory: user: unknown userid 501
```

That is a serious error which prevents use of podman. There are valid cases where HOME may not be defined, for example when running code wrong inside tox, by default HOME is not passed.

There are other ways to retrieve current user home directory than HOME variable and podman should not rely on it, it should have a fallback. I would not be surprised if the same bug would be valid for other platforms.

For example I know how to correctly retrieve home directory with python in a portable way that does not rely on HOME variable but podman is not written in python.

While fixing this we also need to add a regression test, doing an `unset HOME && podman info` should pass.

**Comment by rhatdan** [comment]
This works on a Works on Linux.
@baude PTAL

**Comment by ssbarnea** [comment]
I will need to gather extra info on this, feel free to assign it to me but i will only be able to look more into it next week. I am not yet sure which were all the elements that made podman not work under tox but worked fine outside. 

One I know for sure is that my macos userid is indeed 501 and that HOME was unset.

**Comment by baude** [comment]
yeah, i dont have a full grasp on unset and userids on macos.  any help here is appreciated.  maybe @ashley-cui knows?

**Comment by ashley-cui** [comment]
I'm not entirely sure either probably need to do research on it

**Comment by rhatdan** [comment]
@ssbarnea @baude @ashley-cui Where are we on this one?

**Comment by avadhanij6si** [comment]
Will mention it here since I am seeing the same issue on my end, in a completely different scenario. It happened when using Bazel to build images. I had set Bazel to use `podman` instead of `docker`, and then I saw this as the cause for failure - 

`failed (Exit 1)
time="2021-11-08T00:27:41-08:00" level=error msg="finding config on system: unable to resolve HOME directory: user: unknown userid 501"`

I am not sure if it provides additional insight.
